moon-store-mac-os/moon-store-mac-os/Services/BaseNetworkService.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  BaseNetworkService.swift
+//  moon-store-mac-os
+//
+//  Created by Jose Luna on 12/3/24.
+//
+
+import Foundation
+
+// TODO: - Error handling
+
+private enum Constants {
+    static let scheme: String = "https"
+    static let baseUrl: String = "moon-store-production.up.railway.app"
+}
+
+actor BaseNetworkService {
+    private var components: URLComponents = {
+        var components = URLComponents()
+        components.scheme = Constants.scheme
+        components.host = Constants.baseUrl
+        return components
+    }()
+    
+    private func makeQueryItems(parameters: [String: Any]) -> [URLQueryItem] {
+        parameters.map { key, value in
+            URLQueryItem(name: key, value: value as? String)
+        }
+    }
+    
+    // MARK: - HTTP GET
+    
+    func getData(path: MSPath) async throws -> Data {
+        components.path = path.endpoint
+        
+        guard let url = components.url else {
+            throw NSError(domain: "Invalid url", code: 0)
+        }
+        
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = "GET"
+        
+        return try await request(urlRequest: urlRequest)
+    }
+    
+    // MARK: - HTTP POST
+    
+    func postData(path: MSPath, parameters: [String: Any]) async throws -> Data {
+        components.path = path.endpoint
+        components.queryItems = makeQueryItems(parameters: parameters)
+        
+        guard let url = components.url else {
+            throw NSError(domain: "Invalid url", code: 0)
+        }
+        
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = "POST"
+        
+        let json = try? JSONSerialization.data(withJSONObject: parameters)
```

Comment:
Try opcional?

Reply:
You could notice a `TODO` comment for error handler feature. We will update how we are handling errors in this class to specify context about any error we can get.